pkg/op/context.go

```diff
@@ -7,16 +7,16 @@ import (
 
 type key int
 
-var (
-	issuer key = 0
+const (
+	issuerKey key = 0
 )
 
 type IssuerInterceptor struct {
 	issuerFromRequest IssuerFromRequest
 }
 
-//NewIssuerInterceptor will set the issuer into the context
-//by the provided IssuerFromRequest (e.g. returned from StaticIssuer or IssuerFromHost)
+// NewIssuerInterceptor will set the issuer into the context
+// by the provided IssuerFromRequest (e.g. returned from StaticIssuer or IssuerFromHost)
 func NewIssuerInterceptor(issuerFromRequest IssuerFromRequest) *IssuerInterceptor {
```

Comment:
I've written code similar to this in order to pass things through into my implementation of `Storage`.  How about we simply put the `http.Request` into the `context.Context` that is passed into most `Storage` functions.  That would solve this class of problem for most people.  We can also include the decoded body. 

I'm not a fan of Context, but doing this could solve a whole set of situations all at once.

Reply:
The implementation of this function was not part of this PR. Other changes in the file just triggered `gofmt` on the above comments.

If there is additional functionality you want from the context, I suggest to open an issue and we can discuss there.  Note that is you do need a custom context, you can always provide your own middleware that takes care of that:

https://github.com/zitadel/oidc/blob/6f8a392ce20a6063081438b38a43d09bf74d81dd/pkg/op/op.go#L491